File: 2018/week-1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int h, s;
    do
    {
        h = get_int("Height: "); //asks the user for the pyramid height.
    }
    while (h < 0 || h > 23);
    //Loop through the number of rows of pyramid based on h
    for (int i = 0; i < h; i++)
    {
        //For 'x' row, print out 'h-x' spaces followed by 'x' hashes.
        for (s = 1; s < h - i; s++)
        {
            printf(" ");
        }
        printf("#");
        for (int hash = 0; hash <= h - s; hash++)
        {
            printf("#");
        }
        //At the end, print a new line.
        printf("\n");
    }
}

File: 2019/pset6/mario/less/mario.py
from cs50 import get_int

# prints mario blocks

while(True):
    # ask for half-pyramid's height
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# if it is within the range, print the pyramid
for row in range(height):
    print(" " * (height-row-1), end="")
    print("#" * (row+1), end="")
    print("")